cudamapper/src/cudamapper_utils.hpp

```diff
@@ -49,5 +49,31 @@ namespace cudamapper
 /// \param unfused_overlaps vector of overlaps, sorted by (query_id, target_id) combination and query_start_position
 void fuse_overlaps(std::vector<Overlap>& fused_overlaps, const std::vector<Overlap>& unfused_overlaps);
 
+std::string string_slice(const std::string& s, std::size_t start, std::size_t end);
+
+/// \brief Given a string s, produce its kmers (length <kmer-length>) and return them as a vector of strings.
+/// \param s A string sequence to kmerize.
+/// \param kmer_length A kmer length to use for producing kmers.
+/// \param stride The number of bases to skip when selecting kmers (most often, this should be equal to 1).
+std::vector<std::string> kmerize_string(const std::string& s, int kmer_length, int stride);
+
+/// \brief Given two sorted vectors of comparable types, return a size_t count of the number of shared elements.
+/// \param a A vector of elements. These must be comparable (i.e., they must implement the == and < operators).
+/// \param b A vector of elements. These must be comparable and must be the same type as those in a.
+template <typename T>
+std::size_t count_shared_elements(const std::vector<T>& a, const std::vector<T>& b);
+
+/// \brief Given two sequences 'a' and 'b', calculate an estimate of their similarity
+/// Calculates the approximate nucleotide identity (or "similarity")
+/// estimated from the Jaccard index of the kmers of strings a and b.
+/// Note: This function assumes that a and b are on the same strand; you may need to
+/// reverse-complement one of the strings if testing similarity on strings from different
+/// strands.
+/// \param a A C++ string
+/// \param b A C++ string
+/// \param kmer_length The kmer length to use for estimating similarity.
+/// \param stride The number of bases to stride between kmers.
+float similarity(std::string a, std::string b, int kmer_length, int stride);
```

Comment:
Function name could be a bit more descriptive

Reply:
Addressed in edd521b